app/controllers/SelectTransactionsController.scala

```diff
@@ -48,9 +47,10 @@ class SelectTransactionsController @Inject() (
   def form: Form[CashTransactionDates] = formProvider()
 
   def onPageLoad: Action[AnyContent] = identify.async { implicit request =>
-    for {
-      _ <- cache.clear(request.eori)
-    } yield Ok(view(form))
+    cache.get(request.eori).map {
```

Comment:
Also here, we should have a `recover` clause in case DB interaction fails, then the user will still see the page although we failed to get their cached answers.